File: server/src/main/java/dev/oscarrojas/drawandguess/websocket/protocol/PayloadType.java
package dev.oscarrojas.drawandguess.websocket.protocol;

public enum PayloadType {
    BINARY(0),
    JSON(1);

    final int value;

    PayloadType(int value) {
        this.value = value;
    }

    public static PayloadType fromValue(int value) {
        for (PayloadType type : PayloadType.values()) {
            if (type.value == value) {
                return type;
            }
        }
        throw new IllegalArgumentException("Unknown payload type value: " + value);
    }
}

File: server/src/main/typescript/PayloadType.ts
export enum PayloadType {
  BINARY = 0,
  JSON = 1,
}

export function getPayloadType(value: number): PayloadType {
  if (value === PayloadType.BINARY) {
    return PayloadType.BINARY;
  } else if (value === PayloadType.JSON) {
    return PayloadType.JSON;
  } else throw new Error("Unknown payload type value: " + value);
}
